sections/Animation/Animation.tsx

```diff
@@ -0,0 +1,154 @@
+import { Section } from "deco/blocks/section.ts";
+import { useId } from "$store/sdk/useId.ts";
+
+interface Children {
+  label?: string;
+  section: Section;
+}
+
+interface Props {
+  animationType?:
+    | "fade-in"
+    | "fade-in-bottom"
+    | "slide-left"
+    | "slide-right"
+    | "zoom-in";
+  /**
+   * @default 0.3
+   */
+  duration?: string;
+  children: Children;
+}
+
+function Animation(
+  { children, animationType = "fade-in", duration = "0.3" }: Props,
+) {
+  const { section } = children;
+
+  const { Component, props } = section;
+  const id = useId();
+
+  const animationClasses = {
+    "fade-in": "animate-fade-in",
+    "fade-in-bottom": "animate-fade-in-bottom",
+    "slide-left": "animate-slide-left",
+    "slide-right": "animate-slide-right",
+    "zoom-in": "animate-zoom-in",
+  };
+
+  const animationClass = animationClasses[animationType];
+
+  return (
+    <>
+      <style
+        dangerouslySetInnerHTML={{ __html: animationByType[animationType] }}
+      >
+      </style>
+      <div
+        id={id}
+        class="opacity-0"
+        style={{ animationDuration: duration + "s" }}
+      >
+        <Component {...props} />
+      </div>
+      <script
+        dangerouslySetInnerHTML={{
+          __html: `
+                var observer = new IntersectionObserver(function(entries) {
+                    entries.forEach(entry => {
+                        if (entry.isIntersecting) {
+                            entry.target.classList.add("${animationClass}");
+                            entry.target.classList.remove("opacity-0");
+                            observer.disconnect();
+                        }
+                    });
+                }, { threshold: 0.50 });
+            
+                var element = document.getElementById('${id}');
+                observer.observe(element);
+            `,
+        }}
+      />
+    </>
+  );
+}
+
+export default Animation;
+
+const animationByType = {
```

Comment:
Add this into tailwind.css file. Also, you could add variables to handle animation properties, like tailiwind

Reply:
example of css variables to control the animation:
```css
.animate-fade-in {
  animation: fade-in var(--anim-dur, 1s) ease-out;
}
```

now, you can do something like this:

```tsx
interface Props {
  duration?: number;
}

// ...

<element class={`${animationClass}`} style={duration ? { --anim-dur: duration }: undefined} />
```

Also, you can do something better, with predifined values, via the theme;

```ts
interface Theme {
  timings: "slow" | "2x-slow" | "normal" // etc
}
```
then can create class like:

```css
.animationSlow { --anim-dur: 2000ms }
.animation2xSlow { --anim-dur: 1000ms }
.animationNormal { --anim-dur: 750ms }
```

Reply:
I started using tailwind.css, but I don`t want to insert css (and turn int heaviest) in all new sites that probably won`t use animation, like e-commerce.